SerialThread.py

```diff
@@ -75,45 +81,72 @@ def trySendMessage(self):
-                    if try_ascii_decode(byteList[IDLENGTH]) and byteList[IDLENGTH].decode("ascii") in COM_NAME:
-                        self.sig_print.emit("Failed " + COM_NAME[byteList[IDLENGTH].decode("ascii")])
-                    else:
-                        self.sig_print.emit("Failed 0x" + byteList[IDLENGTH].hex())
-                    del byteList[0:2]
-
-                elif byteList[0] == String_ID:
-                    bytes = bytearray(map(_bytes_to_byte, byteList[1:len(byteList)]))
-                    str = bytes.decode("ascii")
-                    self.sig_print.emit(str)
-                    del byteList[0:len(byteList)]
-
-                elif len(byteList) >= chunkLength and byteList[0] in self.IdSet:
-                    chunk = byteList[0:chunkLength]
-                    data = list(map(_bytes_to_int, chunk))
-                    self.sig_received.emit(data)
-                    del byteList[0:chunkLength]
-
-                elif len(byteList) > chunkLength:
-                    del byteList[0]
-
-                else:
-                    break
+                # if byteList[0] == Good_ID: #Should be byteList[IDLENGTH:chunklength] ??
+                #     if try_ascii_decode(byteList[IDLENGTH]) and byteList[IDLENGTH].decode("ascii") in COM_NAME:
+                #         self.sig_print.emit("Successful " + COM_NAME[byteList[IDLENGTH].decode("ascii")])
+                #     else:
+                #         self.sig_print.emit("Successful 0x" + byteList[IDLENGTH].hex())
+                #     del byteList[0:2]
+                #
+                # elif byteList[0] == Bad_ID:
+                #     if try_ascii_decode(byteList[IDLENGTH]) and byteList[IDLENGTH].decode("ascii") in COM_NAME:
+                #         self.sig_print.emit("Failed " + COM_NAME[byteList[IDLENGTH].decode("ascii")])
+                #     else:
+                #         self.sig_print.emit("Failed 0x" + byteList[IDLENGTH].hex())
+                #     del byteList[0:2]
+                #
+                # elif byteList[0] == String_ID:
+                #     bytes = bytearray(map(_bytes_to_byte, byteList[1:len(byteList)]))
+                #     str = bytes.decode("ascii")
+                #     self.sig_print.emit(str)
+                #     del byteList[0:len(byteList)]
+
+                # calculate length (if possible)
+                try:
+                    data_and_info: Dict[str, int] = RadioController.extract_subpacket(byteList)
+                except ValueError:
+                    del byteList[0:1]
+                    continue
+                except Exception as e:
+                    print(e)
+                    del byteList[0:1]
+                    continue
+
+                chunk_length = data_and_info['length']
+                data = byteList[0:chunk_length]  # convert all of the data in the chunk to byte
+
+                parsed_subpacket_data = RadioController.parse_data( data_and_info['id'], data, chunk_length )
```

Comment:
Would it make more sense if we put this inside of extract_subpacket (or inside another method containing them both) so that way we're not going "in and out" of RadioController twice?

Or would that make things worse? Not sure.